controllers/searchoperator_controller.go

```diff
@@ -180,14 +181,21 @@ func (r *SearchOperatorReconciler) Reconcile(req ctrl.Request) (ctrl.Result, err
 		return ctrl.Result{}, nil
 	}
 	if persistence {
+		expectedSts := r.expectedStatefulSet(r.Client,
+			instance, true, persistence)
 		//If running PVC deployment nothing to do
-		if persistenceStatus == statusUsingPVC && deployStatus == deploy &&
-			isStatefulSetAvailable(r.Client) && r.isPodRunning(true, 1) {
+		if persistenceStatus == statusUsingPVC && deployStatus == deploy && isStatefulSetAvailable(r.Client) &&
```

Comment:
Add a comment here Please - on what all cases we dont need to go anything

Reply:
Done